elfpy/utils/logs.py

```diff
@@ -23,81 +24,83 @@ def setup_logging(
-        log_file_and_stdout can be set to True to log to both file and standard output (console). If
-        neither log_filename nor log_file_and_stdout is specified, the log messages will be sent to
-        standard output only.
+    The log_filename can be a path to the log file. If log_filename is not provided,
+    log_file_and_stdout can be set to True to log to both file and standard output (console). If
+    neither log_filename nor log_file_and_stdout is specified, the log messages will be sent to
+    standard output only.
 
     Arguments
-    ----------
-        log_filename : (str, optional)
-            Path and name of the log file.
-        max_bytes : (int, optional)
-            Maximum size of the log file in bytes. Defaults to elfpy.DEFAULT_LOG_MAXBYTES.
-        log_level : (int, optional)
-            Log level to track. Defaults to elfpy.DEFAULT_LOG_LEVEL.
-        delete_previous_logs : (bool, optional)
-            Whether to delete previous log file if it exists. Defaults to False.
-        log_stdout : (bool, optional)
-            Whether to log to standard output. Defaults to True.
-        log_format_string : (str, optional)
-            Log formatter object. Defaults to None.
+    ---------
+    log_filename : str, optional
+        Path and name of the log file.
+    max_bytes : int, optional
+        Maximum size of the log file in bytes. Defaults to elfpy.DEFAULT_LOG_MAXBYTES.
+    log_level : int, optional
+        Log level to track. Defaults to elfpy.DEFAULT_LOG_LEVEL.
+    delete_previous_logs : bool, optional
+        Whether to delete previous log file if it exists. Defaults to False.
+    log_stdout : bool, optional
+        Whether to log to standard output. Defaults to True.
+    log_format_string : str, optional
+        Log formatter object. Defaults to None.
+    keep_previous_handlers : bool, optional
+        Whether to keep previous handlers. Defaults to False.
 
     .. todo::
         - Fix the docstring
         - Test the various optional input combinations
     """
-    # handle defaults
-    if max_bytes is None:
-        max_bytes = elfpy.DEFAULT_LOG_MAXBYTES
-    if log_level is None:
-        log_level = elfpy.DEFAULT_LOG_LEVEL
-    if log_format_string is None:
-        log_formatter = logging.Formatter(elfpy.DEFAULT_LOG_FORMATTER, elfpy.DEFAULT_LOG_DATETIME)
-    else:
-        log_formatter = logging.Formatter(log_format_string, elfpy.DEFAULT_LOG_DATETIME)
-    # create log handlers
-    handlers = logging.getLogger().handlers if keep_previous_handlers else []
-    # pipe to stdout if requested
-    if log_stdout:
-        stream_handler = logging.StreamHandler(sys.stdout)
-        stream_handler.setFormatter(log_formatter)
-        handlers.append(stream_handler)
-    # log to file
+    # pylint: disable=too-many-arguments
+    # sane defaults to avoid spam from libraries
+    logging.getLogger("urllib3").setLevel(logging.WARNING)
```

Comment:
```    logging.getLogger("urllib3").setLevel(logging.WARNING)
    logging.getLogger("web3").setLevel(logging.WARNING)
    warnings.filterwarnings("ignore", category=UserWarning, module="web3.contract.base_contract")```
    These should not happen here -- they should be in whatever script is being run. We want to be as unopinionated as possible about logging. Changing package levels "under the hood" is bad.

Reply:
right, i'll move it to main.py, that's where I had it initially!